Updated AWS-SDK-PHP version

Rockburner opened the pull request:
Updated AWS-SDK-PHP dependency to 3.* because the library is bug-fixed too fast for us to keep up with.

Rockburner committed:
Updated AWS-SDK-PHP version

Updated AWS-SDK-PHP dependency to 3.* because the library is bug-fixed too fast for us to keep up with.

Rockburner committed:
Updated version of irap/emailers

Another dependency that's out of date. :(

Rockburner commented:
Updated irap/emailers as well.

iRAP-Stu commented:
I wrote a quick script to test the aws emailer and the phpmailer worked for me using this version, in php 8.2. However, it would be good to add the requirement that one needs the `ext-xml` in the package.json requirements section, because if one does not have it, then one gets the following error message:

```
PHP Fatal error:  Uncaught Error: Class "SimpleXMLElement" not found in /path/to/project/vendor/aws/aws-sdk-php/src/Api/Parser/PayloadParserTrait.php:39
Stack trace:
#0 /path/to/project/vendor/aws/aws-sdk-php/src/Api/Parser/QueryParser.php(44): Aws\Api\Parser\QueryParser->parseXml()
#1 /path/to/project/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php(126): Aws\Api\Parser\QueryParser->__invoke()
#2 /path/to/project/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php(93): Aws\WrappedHttpHandler->parseResponse()
#3 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(204): Aws\WrappedHttpHandler->Aws\{closure}()
#4 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler()
#5 /path/to/project/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#6 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Promise\TaskQueue->run()
#7 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
#8 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
#9 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
#10 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
#11 /path/to/project/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(59): GuzzleHttp\Promise\Promise->wait()
#12 /path/to/project/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(78): Aws\AwsClient->execute()
#13 /path/to/project/vendor/irap/aws-emailer/src/AwsEmailer.php(107): Aws\AwsClient->__call()
#14 /path/to/project/main.php(41): iRAP\AwsEmailer\AwsEmailer->send()
#15 {main}
```

Rockburner commented:
> I wrote a quick script to test the aws emailer and the phpmailer worked for me using this version, in php 8.2. However, it would be good to add the requirement that one needs the `ext-xml` in the package.json requirements section, because if one does not have it, then one gets the following error message:
> 
> ```
> PHP Fatal error:  Uncaught Error: Class "SimpleXMLElement" not found in /path/to/project/vendor/aws/aws-sdk-php/src/Api/Parser/PayloadParserTrait.php:39
> Stack trace:
> #0 /path/to/project/vendor/aws/aws-sdk-php/src/Api/Parser/QueryParser.php(44): Aws\Api\Parser\QueryParser->parseXml()
> #1 /path/to/project/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php(126): Aws\Api\Parser\QueryParser->__invoke()
> #2 /path/to/project/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php(93): Aws\WrappedHttpHandler->parseResponse()
> #3 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(204): Aws\WrappedHttpHandler->Aws\{closure}()
> #4 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler()
> #5 /path/to/project/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
> #6 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Promise\TaskQueue->run()
> #7 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
> #8 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
> #9 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
> #10 /path/to/project/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
> #11 /path/to/project/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(59): GuzzleHttp\Promise\Promise->wait()
> #12 /path/to/project/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(78): Aws\AwsClient->execute()
> #13 /path/to/project/vendor/irap/aws-emailer/src/AwsEmailer.php(107): Aws\AwsClient->__call()
> #14 /path/to/project/main.php(41): iRAP\AwsEmailer\AwsEmailer->send()
> #15 {main}
> ```

iRAP-Stu committed:
updates composer.json to specify that the package requires the XML PHP
extension, because otherwise you get an exception to do with not finding
the SimpleXML class when sending an email through the AWS emailer

iRAP-Stu committed:
Merge branch 'master' into aws-sdk-php
